### PR TITLE
Add docstrings for `conj`, `real` and `imag` on arrays

### PR DESCRIPTION
**Pull request by nalimilan**
These were only documented for numbers.

### CODE REVIEW

**Comment by dkarrasch** [comment]
Are they exposed automatically in the online documentation, or do we need to add them there as well?

**Comment by nalimilan** [comment]
Good point. For some reason, for `real` only the method for `::Complex` was included in the manual. The existing docstring for `::Type` wasn't even listed anywhere AFAICT.

File: base/arraymath.jl [hunk 1]
```diff
@@ -26,9 +26,100 @@ julia> A
...
-for f in (:-, :conj, :real, :imag)
-    @eval ($f)(A::AbstractArray) = broadcast_preserving_zero_d($f, A)
-end
+"""
+    conj(A::AbstractArray)
+
+Return an array containing the complex conjugate of each entry in array `A`.
+
+Equivalent to `conj.(A)`, except when `A` is a 0-dimensional result,
+in which case a 0-dimensional container is returned (rather than a scalar).
```
Comment (mcabbott) [inline base/arraymath.jl:35]: Another difference is that these don't always copy:
```
julia> a = [1,2,3]; b = real(a); b[1] = 99; a
3-element Vector{Int64}:
 99
  2
  3
```

File: base/arraymath.jl [hunk 2]
```diff
@@ -26,9 +26,100 @@ julia> A
...
+```jldoctest
+julia> A = [1+im 2-im; 2+2im 3+im]
+2×2 Matrix{Complex{Int64}}:
+ 1+1im  2-1im
+ 2+2im  3+1im
+
+julia> conj(A)
+2×2 Matrix{Complex{Int64}}:
+ 1-1im  2+1im
+ 2-2im  3-1im
```
Comment (mcabbott) [inline base/arraymath.jl:47]: Could we make the examples more compact without loss? E.g.
```suggestion
julia> conj([1 2im; 3 4im])
2×2 Matrix{Complex{Int64}}:
 1+0im  0-2im
 3+0im  0-4im
```

**Comment by mcabbott** [comment]
Might be worth also including `reim`:
```
julia> reim([1, 2im, 3])
([1, 0, 3], [0, 2, 0])

julia> reim(fill(1+2im))
(fill(1), fill(2))
```
I see that `isreal` just includes the example in the basic docstring
```
  julia> isreal([4.; complex(0,1)])
  false
  ```

File: base/arraymath.jl (hunk 2 above)
Comment (nalimilan) [reply base/arraymath.jl:47]: I just copied the existing example for `conj!`. I'd say it makes sense to print `A` as it makes it very easy to compare the two arrays.

Comment (mcabbott) [reply base/arraymath.jl:47]: Fair point. I do wonder if they could all be made less of a visual puzzle, `2-im` vs `2+2im` etc. 

Do they need to be matrices? Something like `conj([1, 2im, 3 + 4im])` would be easy to compare.

File: base/arraymath.jl (hunk 1 above)
Comment (nalimilan) [reply base/arraymath.jl:35]: Good catch. I've also noticed that the code is poorly organized, with methods scatters without any reason across abstractarraymath.jl and arraymath.jl. I'll regroup them.

File: base/arraymath.jl (hunk 2 above)
Comment (nalimilan) [reply base/arraymath.jl]: OK let's use a vector.

**Comment by nalimilan** [comment]
Good to go?

**Comment by nalimilan** [comment]
I'll merge tomorrow barring objections.

**Pull request merged**